internal/pkg/expanders/graph.go

```diff
@@ -0,0 +1,595 @@
+			Name:                  "Apps",
+			Display:               "Apps",
+			ItemType:              itemTypeAppList,
+			ExpandURL:             "/applications",
+			SuppressSwaggerExpand: true,
+			SuppressGenericExpand: true,
+		},
+		{
+			Parentid:              currentItem.ID,
+			ID:                    "/serviceprincipals",
+			Namespace:             namespace,
+			Name:                  "Service Principals",
+			Display:               "Service Principals",
+			ItemType:              itemTypeSpList,
+			ExpandURL:             "/servicePrincipals", //&$search="displayName:damoo"
+			SuppressSwaggerExpand: true,
+			SuppressGenericExpand: true,
+		},
+	}
+
+	return ExpanderResult{
+		Err:               nil,
+		SourceDescription: "GraphExpander request",
+		Nodes:             baseItems,
+		IsPrimaryResponse: false,
+	}
+}
+
+func (e *GraphExpander) showMe(ctx context.Context, currentItem *TreeNode) ExpanderResult {
+
+	// make the graph call
+	data, err := e.armClient.DoRequest(ctx, "GET", currentItem.ExpandURL)
+	if err == nil {
+		var meResponse MeResponse
+		jsonErr := json.Unmarshal([]byte(data), &meResponse)
+		if jsonErr == nil {
+			return ExpanderResult{
+				Response:          ExpanderResponse{Response: data, ResponseType: interfaces.ResponseJSON},
+				SourceDescription: "GraphMeExpander request",
+				Nodes:             []*TreeNode{},
+				IsPrimaryResponse: true,
+			}
+		}
+		return ExpanderResult{
+			SourceDescription: "GraphExpander failed to map ME result to object",
+			Err:               jsonErr,
+		}
+	}
+
+	return ExpanderResult{
+		SourceDescription: "GraphExpander failed to show ME",
+		Err:               err,
+	}
+}
+
+func (e *GraphExpander) listAppsOrSps(ctx context.Context, currentItem *TreeNode, queryText string) ExpanderResult {
+	// get the list of apps
+	url := currentItem.ExpandURL
+	expandURLRoot := currentItem.ID
+
+	if queryText != "" { // are we in search mode?
+		expandURLRoot = currentItem.Parent.ID
+		url = expandURLRoot + "?$search=\"displayName:" + queryText + "\""
+	}
+
+	data, err := e.armClient.DoRequest(ctx, "GET", url)
+	if err == nil {
+		return makeAppsList(data, expandURLRoot, currentItem)
+	}
+
+	return ExpanderResult{
+		SourceDescription: "GraphExpander failed to list apps or service principals",
+		Err:               err,
+	}
+}
+
+func (e *GraphExpander) listFilteredApps(ctx context.Context, currentItem *TreeNode, url string) ExpanderResult {
+	// get the list of apps
+	expandURLRoot := currentItem.Parent.ID
+
+	data, err := e.armClient.DoRequest(ctx, "GET", url)
+	if err == nil {
+		return makeAppsList(data, expandURLRoot, currentItem)
+	}
+
+	return ExpanderResult{
+		SourceDescription: "GraphExpander failed to list apps owned by ME",
+		Err:               err,
+	}
+}
+
+func (e *GraphExpander) getAppByID(ctx context.Context, currentItem *TreeNode) ExpanderResult {
+	commandChannel := make(chan string, 1)
+	commandPanelNotification := func(state interfaces.CommandPanelNotification) {
```

Comment:
Optional: This method seems pretty useful maybe a good candidate to move under the `commandPanel` for reused. 

So `appID := <-commandChannel` would become `appID := <- e.commandPanel.CompletedInput()` or something